diff --git a/src/RuleCodeSamplePrinter/PHPStanRuleCodeSamplePrinter.php b/src/RuleCodeSamplePrinter/PHPStanRuleCodeSamplePrinter.php
index 7b825cd..b14c5b2 100644
--- a/src/RuleCodeSamplePrinter/PHPStanRuleCodeSamplePrinter.php
+++ b/src/RuleCodeSamplePrinter/PHPStanRuleCodeSamplePrinter.php
@@ -4,6 +4,8 @@ declare(strict_types=1);
 
 namespace Symplify\RuleDocGenerator\RuleCodeSamplePrinter;
 
+use Nette\Utils\Strings;
+use ReflectionClass;
 use Symplify\RuleDocGenerator\Contract\CodeSampleInterface;
 use Symplify\RuleDocGenerator\Contract\RuleCodeSamplePrinterInterface;
 use Symplify\RuleDocGenerator\Printer\CodeSamplePrinter\BadGoodCodeSamplePrinter;
@@ -46,15 +48,14 @@ final class PHPStanRuleCodeSamplePrinter implements RuleCodeSamplePrinterInterfa
         RuleDefinition $ruleDefinition
     ): array {
         $lines = [];
+        $ruleName = Strings::firstLower((new ReflectionClass($ruleDefinition->getRuleClass()))->getShortName());
 
         $phpStanNeon = [
-            'services' => [
-                [
-                    'class' => $ruleDefinition->getRuleClass(),
-                    'tags' => ['phpstan.rules.rule'],
-                    'arguments' => $configuredCodeSample->getConfiguration(),
-                ],
-            ],
+            'parameters' => [
+               'silverstan' => [
+                    explode('Rule', $ruleName)[0] => $configuredCodeSample->getConfiguration(),
+                ]
+             ],
         ];
 
         $printedNeon = $this->neonPrinter->printNeon($phpStanNeon);
